src/libpmemfile-posix/truncate.c

```diff
@@ -89,6 +93,11 @@ vinode_truncate(PMEMfilepool *pfp, struct pmemfile_vinode *vinode,
 			TX_SET_DIRECT(inode, mtime, tm);
 			TX_SET_DIRECT(inode, ctime, tm);
 		}
+
+		if (inode->allocated_space != allocated_space) {
+			TX_ADD_DIRECT(&inode->allocated_space);
```

Comment:
I guess these TX_ADD_DIRECT + set pairs could be changes to TX_SET_DIRECT instances.